SubmittingPatches.rst

```diff
@@ -2,32 +2,36 @@
 Submitting Patches to Ceph
 ==========================
 
-This is based on Documentation/SubmittingPatches from the Linux kernel,
-but has pared down significantly and updated based on the Ceph project's
-best practices.
+Patches to Ceph can be divided into three categories:
 
-The patch signing procedures and definitions are unmodified.
+    1. patches targeting Ceph kernel code
+    2. patches targeting the "master" branch
+    3. patches targeting stable branches (e.g.: "nautilus")
 
+Some parts of Ceph - notably the RBD and CephFS kernel clients - are maintained
+within the Linux Kernel. For patches targeting this code, please refer to the
+file ``SubmittingPatches-kernel.rst``.
 
-SIGNING CONTRIBUTIONS
-=====================
+The rest of this document assumes that your patch relates to Ceph code that is
+maintained in the GitHub repository https://github.com/ceph/ceph
 
-In order to keep the record of code attribution clean within the source
-repository, follow these guidelines for signing patches submitted to the
-project. These definitions are taken from those used by the Linux kernel
-and many other open source projects.
+If you have a patch that fixes an issue, feel free to open a GitHub pull request
+("PR") targeting the "master" branch, but do read this document first, as it
+contains important information for ensuring that your PR passes code review
+smoothly.
 
+For patches targeting stable branches (e.g. "nautilus"), please also see
+the file ``SubmittingPatches-backports.rst``.
+
+.. contents::
+   :depth: 3
 
-1. Sign your work
------------------
 
-To improve tracking of who did what, especially with patches that can
-percolate to their final resting place in the kernel through several
-layers of maintainers, we've introduced a "sign-off" procedure on
-patches that are being emailed around.
+Sign your work
```

Comment:
Perhaps you'd like to suggest/indicate what needs to go in the headers of new files if one creates them... I usually just borrow some content of a file next to it in the directory. 
But that is not always fitting.
I'm thinking that there should be something there in the lines that the file falls under the same copyright as the project.
Also what to do with the original author. Because I'm borrowing, most of the attribution goes to Dreamhost or RedHat.